### PR TITLE
Sprite Fix

### PR DESCRIPTION
**Pull request by infraredbaron**
Deagle, Sky Driver, Colt and Deckard have proper in-hands and updated sprites

## About The Pull Request

image

## Changelog
:cl:
imageadd: Fitting in-hands for deagle, sky driver, colt and deckard;
code: Added slide code to deagle
/:cl:

### CODE REVIEW

**Comment by ArmolitskiyCarpoeb** [comment]
За что Sky Driver?! Теперь он какой-то противный револьвер с хуевиной.

**Comment by maikilangiolo** [comment]
merge it already

File: code/modules/projectiles/guns/projectile/pistol/avasarala.dm [hunk 1]
```diff
@@ -26,3 +26,17 @@
...
+
+/obj/item/weapon/gun/projectile/avasarala/update_icon()
+	..()
+
+	var/iconstring = initial(icon_state)
+
+	if (!ammo_magazine || !length(ammo_magazine.stored_ammo))
+		iconstring += "_slide"
+
+	icon_state = iconstring
```
Comment (AlexMorgan3817) [inline code/modules/projectiles/guns/projectile/pistol/avasarala.dm:38]: #5940

File: code/modules/projectiles/guns/projectile/pistol/avasarala.dm [hunk 2]
```diff
@@ -26,3 +26,17 @@
 
 	price_tag = 1600
 	spawn_tags = SPAWN_TAG_FS_PROJECTILE
+
+/obj/item/weapon/gun/projectile/avasarala/update_icon()
```
Comment (hyperioo) [inline code/modules/projectiles/guns/projectile/pistol/avasarala.dm:30]:
```suggestion
/obj/item/weapon/gun/projectile/avasarala/on_update_icon()
```
To be consistent with #5940

File: code/modules/projectiles/guns/projectile/pistol/avasarala.dm [hunk 3]
```diff
@@ -26,3 +26,17 @@
...
+	var/iconstring = initial(icon_state)
+
+	if (!ammo_magazine || !length(ammo_magazine.stored_ammo))
+		iconstring += "_slide"
+
+	icon_state = iconstring
+
+/obj/item/weapon/gun/projectile/avasarala/Initialize()
+	. = ..()
+	update_icon()
```
Comment (hyperioo) [inline code/modules/projectiles/guns/projectile/pistol/avasarala.dm:42]:
```suggestion
	update_icon()

```
Add newline at end of file

**Review by hyperioo — CHANGES_REQUESTED** [review]

**Review by hyperioo — APPROVED** [review]

File: code/modules/projectiles/guns/projectile/pistol/avasarala.dm [hunk 4]
```diff
@@ -26,3 +26,17 @@
...
+
+/obj/item/weapon/gun/projectile/avasarala/on_update_icon()
+	..()
+
+	var/iconstring = initial(icon_state)
+
+	if (!ammo_magazine || !length(ammo_magazine.stored_ammo))
+		iconstring += "_slide"
+
+	icon_state = iconstring
```
Comment (hyperioo) [inline code/modules/projectiles/guns/projectile/pistol/avasarala.dm:38]:
```suggestion
	SetIconState(iconstring)
```

**Review by hyperioo — CHANGES_REQUESTED** [review]
Forgot this change.

**Review by hyperioo — APPROVED** [review]

**Pull request merged**